torch/csrc/jit/tensorexpr/ir_cloner.cpp

```diff
@@ -0,0 +1,343 @@
+    params_new.push_back(param->accept_mutator(this));
+  }
+  return new Intrinsics(v->op_type(), params_new);
+}
+
+Expr* IRCloner::mutate(Term* v) {
+  Expr* scalar_new = v->scalar()->accept_mutator(this);
+
+  std::vector<Expr*> variables_new;
+  variables_new.reserve(v->variables().size());
+  for (auto* t : v->variables()) {
+    variables_new.push_back(t->accept_mutator(this));
+  }
+  return new Term(v->hasher(), scalar_new, variables_new);
+}
+
+Expr* IRCloner::mutate(Polynomial* v) {
+  Expr* scalar_new = v->scalar()->accept_mutator(this);
+
+  std::vector<Term*> variables_new;
+  variables_new.reserve(v->variables().size());
+  for (auto* t : v->variables()) {
+    variables_new.push_back(static_cast<Term*>(t->accept_mutator(this)));
+  }
+  return new Polynomial(v->hasher(), scalar_new, variables_new);
+}
+
+Expr* IRCloner::mutate(RoundOff* v) {
+  return new RoundOff(
+      v->lhs()->accept_mutator(this), v->rhs()->accept_mutator(this));
+}
+
+Expr* IRCloner::mutate(MaxTerm* v) {
+  Expr* scalar_new = v->scalar() ? v->scalar()->accept_mutator(this) : nullptr;
+
+  std::vector<Expr*> variables_new;
+  variables_new.reserve(v->variables().size());
+  for (auto* t : v->variables()) {
+    variables_new.push_back(t->accept_mutator(this));
+  }
+  return new MaxTerm(
+      v->hasher(), scalar_new, v->propagate_nans(), variables_new);
+}
+
+Expr* IRCloner::mutate(MinTerm* v) {
+  Expr* scalar_new = v->scalar() ? v->scalar()->accept_mutator(this) : nullptr;
+
+  std::vector<Expr*> variables_new;
+  variables_new.reserve(v->variables().size());
+  for (auto* t : v->variables()) {
+    variables_new.push_back(t->accept_mutator(this));
+  }
+  return new MinTerm(
+      v->hasher(), scalar_new, v->propagate_nans(), variables_new);
+}
+
+Expr* IRCloner::mutate(ReduceOp* v) {
+  Expr* body_new = v->body()->accept_mutator(this);
+
+  std::vector<Var*> reduce_args_new;
+  reduce_args_new.reserve(v->reduce_args().size());
+  for (auto* r : v->reduce_args()) {
+    reduce_args_new.push_back(static_cast<Var*>(r->accept_mutator(this)));
+  }
+
+  return new ReduceOp(body_new, reduce_args_new, v->reducer());
+}
+
+Stmt* IRCloner::mutate(For* v) {
+  auto start_new = v->start()->accept_mutator(this);
+  auto stop_new = v->stop()->accept_mutator(this);
+  auto body_new = v->body()->accept_mutator(this);
+
+  return new For(v->var(), start_new, stop_new, body_new, v->loop_options());
+}
+
+Stmt* IRCloner::mutate(Block* v) {
+  std::vector<Stmt*> stmts_new;
+  stmts_new.reserve(v->nstmts());
+  for (Stmt* stmt : *v) {
+    stmts_new.push_back(stmt->accept_mutator(this));
+  }
+  return new Block(stmts_new);
+}
+
+Stmt* IRCloner::mutate(Store* v) {
+  std::vector<Expr*> indices_new;
+  indices_new.reserve(v->indices().size());
+  for (auto ind : v->indices()) {
+    indices_new.push_back(ind->accept_mutator(this));
+  }
+  auto value_new = v->value()->accept_mutator(this);
+  return new Store(v->buf(), indices_new, value_new);
```

Comment:
We have a mutator for Buf, but we're not calling it here. Is that intentional? In what situations are we cloning the Buf?

Reply:
Yeah, I missed that. Fixed now.